doc/adrs/183_add_support_for_project_dependencies.adoc

```diff
@@ -0,0 +1,40 @@
+= [ADR-183] Add support for project dependencies
+
+
+== Context
+
+https://github.com/eclipse-sirius/sirius-web/blob/master/doc/iterations/2025.2/add_support_for_the_import_of_libraries.adoc[This shape] describes how project can import libraries, but does not define how the project-to-library dependency is stored.
+
+We need to capture this information to implement future library import/upgrade strategies.
+For example: we need to know whether a part of a project comes from a read-only library (to prevent users from editing it), or what are the libraries imported by a project that can be upgraded to a newer version.
+
+== Decision
+
+A new table `project_dependency` with the following columns should be created
+
+- `projectId`
+- `libraryId`
+- `readOnly`: a boolean indicating whether the imported library is read-only or not
```

Comment:
Why would that be necessary? If it's not readonly then it should be part of the semantic data and thus modifiable. How could we have a dependency (so something that is not under our responsibility) that we can modify? Either something is in our semantic data (we can modify it, it's ours) or it's in a library (we can't modify it)